Classes/Compiler/BreadcrumbsCompiler.php

```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types = 1);
+namespace B13\Menus\Compiler;
+
+/*
+ * This file is part of TYPO3 CMS-based extension "menus" by b13.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ */
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+class BreadcrumbsCompiler extends AbstractMenuCompiler
+{
+    /**
+     * @inheritDoc
+     */
+    public function compile(ContentObjectRenderer $contentObjectRenderer, array $configuration): array
+    {
+        $cacheIdentifier = $this->generateCacheIdentifierForMenu('breadcrumb', $configuration);
+
+        $excludePages = $contentObjectRenderer->stdWrap($configuration['excludePages'] ?? '', $configuration['excludePages.']);
+        $excludePages = GeneralUtility::trimExplode(',', $excludePages);
+
+        $cacheIdentifier .= '-' . GeneralUtility::shortMD5(json_encode([$excludePages]));
+
+        return $this->cache->get($cacheIdentifier, function () use ($contentObjectRenderer, $configuration, $excludePages) {
+            $pages = $this->menuRepository->getBreadcrumbsMenu($GLOBALS['TSFE']->rootLine, $configuration);
```

Comment:
Adding an extra cache for something that is already available (that is - in "TSFE->rootLine") seems superfluous as we would need to write and read cache entries. I think it's faster to do the exclude pages functionality directly without the cache->get.

Reply:
Hi Benny,